generated-docs/FRP/Event.md

```diff
@@ -36,87 +54,95 @@ Alternative Event
 never :: forall a. Event a
 ```
 
-#### `fold`
+#### `filter`
 
 ``` purescript
-fold :: forall a b. (a -> b -> b) -> Event a -> b -> Event b
+filter :: forall a. (a -> Boolean) -> Event a -> Event a
 ```
 
-Fold over values received from some `Event`, creating a new `Event`.
+Create an `Event` which only fires when a predicate holds.
 
-#### `count`
+#### `subscribe`
 
 ``` purescript
-count :: forall a. Event a -> Event Int
+subscribe :: forall eff a r. Event a -> (a -> Eff (frp :: FRP | eff) r) -> Eff (frp :: FRP | eff) Unit
 ```
 
-Count the number of events received.
+Subscribe to an `Event` by providing a callback.
 
-#### `folded`
+#### `create`
 
 ``` purescript
-folded :: forall a. Monoid a => Event a -> Event a
+create :: forall eff a. Eff (frp :: FRP | eff) { event :: Event a, push :: a -> Eff (frp :: FRP | eff) Unit }
 ```
 
-Count the number of events received.
-
-#### `withLast`
+Create an event and a function which supplies a value to that event.
 
-``` purescript
-withLast :: forall a. Event a -> Event { now :: a, last :: Maybe a }
-```
 
-Compute differences between successive event values.
+### Re-exported from FRP.Event.Class:
 
-#### `filter`
+#### `IsEvent`
 
 ``` purescript
-filter :: forall a. (a -> Boolean) -> Event a -> Event a
+class (Alternative event) <= IsEvent event  where
+  fold :: forall a b. (a -> b -> b) -> event a -> b -> event b
+  mapMaybe :: forall a b. (a -> Maybe b) -> event a -> event b
```

Comment:
Is this a definite move from `Filterable`?  e.g. `Alternative <= Filterable, IsEvent`?

Reply:
No, we can still consider `Filterable` before the next major release. I was hoping you could rebase it on top of this.

Reply:
For sure :) Are you happy with the idea of it being a pre-requisite of `Alternative`?

Reply:
I was thinking something like `class (Alternative event, Filterable event) <= IsEvent event` would make sense.

Reply:
Yep, that - sorry, I'll never get my arrows the right way round, hah! Well, I think this all looks great, and I can't wait to play with it!